### PR TITLE
Add new Super Resolution task to Flutter frontend

### PR DESCRIPTION
**Pull request by anhappdev**
Part of #595

We will need to update the paths to dataset and models later, which is tracked in the issue https://github.com/mlcommons/mobile_app_open/issues/610.

<details>
<summary>Click to show app screenshots!</summary>
 
IMG_8422 Medium
IMG_8425 Medium

IMG_8423 Medium
IMG_8424 Medium

</details>

### CODE REVIEW

File: flutter/lib/l10n/app_en.arb [hunk 1]
```diff
@@ -87,14 +87,17 @@
...
   "benchInfoLanguageProcessingDesc": "Question Answering finds the  best answer to an input question based on a body of text. The reference model, MobileBERT, is evaluated on the Stanford Question Answering Dataset (SQUAD) v1.1 Dev and achieves an F1-score of 90.0.\n\nMobileBERT is a thin, mobile-optimized version of the larger BERT (BERT_LARGE) network, equipped with bottleneck structures and a carefully designed balance between self-attention and feed-forward networks. Like BERT, it is task-agnostic, applicable for a variety of downstream NLP tasks, but for MLPerf we use the variant fine-tuned for Q&A.",
+  "benchInfoSuperResolutionDesc": "Super Resolution tbd.",
```
Comment (anhappdev) [inline flutter/lib/l10n/app_en.arb:100]: @willc2010 @mkankana 
Can someone please provide me a description text for the Super Resolution task?

**Comment by anhappdev** [comment]
@kcwu I am not sure why I cannot add you as reviewer, so comment it here.

File: flutter/cpp/datasets/snu_sr.cc [hunk 2]
```diff
@@ -192,17 +192,20 @@ std::vector<uint8_t> SNUSR::ProcessOutput(const int sample_idx,
...
+
 float SNUSR::ComputeAccuracy() {
   if (ground_truth_list_.empty()) {
-    return 0.0;
+    return -1.0f;
   }
 
-  return psnr_ / ground_truth_list_.size();
+  // Frontend expects normalized accuracy between 0 and 1
+  return psnr_ / ground_truth_list_.size() / 100;
```
Comment (mohitmundhragithub) [inline flutter/cpp/datasets/snu_sr.cc:203]: Accuracy gets rounded off to 2 places of decimal. e.g. 33.45dB becomes 33dB. Any way to keep accuracy upto 4 decimal places?

File: flutter/lib/l10n/app_en.arb (hunk 1 above)
Comment (AasheeshT) [reply flutter/lib/l10n/app_en.arb:100]: "Super Resolution reconstructs a higher resolution(HR) image from lower resolution(LR) image. The reference model EDSR, is evaluated on OpenSR100 dataset."

**Review by AasheeshT — APPROVED** [review]

**Comment by AasheeshT** [comment]
There seems to be a small delta in previous accuracy scores obtained with SNU's python script and this implementation. Do other's also observe it?

**Comment by freedomtan** [comment]
> There seems to be a small delta in previous accuracy scores obtained with SNU's python script and this implementation. Do other's also observe it?

Yes, there are difference between the dataset implementation and what reported by SNU. Mostly, it's caused by
1. image conversion, 
2. what are used to compute PSNR, or
3. different floating point implementations

We can come back this when newer fp32 tflite model is ready and test it with[ TensorFlow's PSNR](https://www.tensorflow.org/api_docs/python/tf/image/psnr), simple python PSNR, and the PSNR in this implementation.

**Comment by freedomtan** [comment]
confirming that it works on some MTK devices with something like

```diff
diff --git a/mobile_back_tflite/cpp/backend_tflite/neuron/tflite_settings_mtk.h b/mobile_back_tflite/cpp/backend_tflite/neuron/tflite_settings_mtk.h
index d67be1b..fd9f3ad 100644
--- a/mobile_back_tflite/cpp/backend_tflite/neuron/tflite_settings_mtk.h
+++ b/mobile_back_tflite/cpp/backend_tflite/neuron/tflite_settings_mtk.h
@@ -86,6 +86,16 @@ benchmark_setting {
   model_checksum: "b7a7620b8b818d64305b51ab796bfb1d"
 }
 
+benchmark_setting {
+  benchmark_id: "super_resolution"
+  accelerator: "neuron"
+  accelerator_desc: "MediaTek NN accelerator via the Neuron Delegate"
+  framework: "TFLite"
+  model_path: "https://github.com/freedomtan/some_super_resolution_tflite_models/raw/main/snu_edsr/tflite/pl_f32b5_fint8.tflite"
+  model_checksum: "a6a9739bf2e983388900eaec62cfd2dd"
+}
+
+
 )SETTINGS";
 
 #endif
```

**Comment by freedomtan** [comment]
The model is likely to be either f64b5 or f32b5, not f28 ones though.

**Comment by mohitmundhragithub** [comment]
> The model is likely to be either f64b5 or f32b5, not f28 ones though. Anyway, we can change it later

William mentioned last week that F32b5 has been finalised. Is that not the case?

**Comment by willc2010** [comment]
@freedomtan , Mohit is right. Based on the voting result last week, the group will use F32b5 for v3.0 submission.

**Pull request merged**